lib/ts/recipe/thirdparty/providers/apple.ts

```diff
@@ -27,7 +27,7 @@ type TypeThirdPartyProviderAppleConfig = {
     };
     scope?: string[];
     authorisationRedirect?: {
-        params?: { [key: string]: string | ((request: Request) => string) };
+        params?: { [key: string]: string | ((request: BaseRequest) => Promise<string>) };
```

Comment:
This is a change in the user facing API? Maybe you want to give the original `req` object here instead of `BaseRequest`?